main.py

```diff
@@ -0,0 +1,69 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import Optional, List
+
+import schemas
+from crud import (
+    get_all_authors,
+    create_author,
+    get_author_detail,
+    get_all_books,
+    create_book,
+)
+from database import get_db
+from settings import API_PREFIX
+
+app = FastAPI()
+
+
+def raise_not_found_exception(instance, id: int):
+    if instance is None:
+        raise HTTPException(
+            status_code=404,
+            detail=f"Item with id {id} not found!"
+        )
+
+
+@app.get(API_PREFIX + "authors/", response_model=List[schemas.Author])
+def read_all_authors(
+        skip: Optional[int] = None,
+        limit: Optional[int] = None,
+        db: Session = Depends(get_db)
+) -> List[schemas.Author]:
+    return get_all_authors(db=db, skip=skip, limit=limit)
+
+
+@app.post(API_PREFIX + "authors/", response_model=schemas.Author)
+def create_author(
```

Comment:
The function `create_author` is defined here, but it conflicts with the imported `create_author` function from `crud.py`. This will lead to a `NameError` when trying to use the function. Consider renaming one of the functions to avoid this conflict.